src/librustdoc/html/markdown.rs

```diff
@@ -68,49 +67,60 @@ const HOEDOWN_EXTENSIONS: libc::c_uint =
 type hoedown_document = libc::c_void;  // this is opaque to us
```

Comment:
The actual representation or whether this can be constructed isn't so much the problem here, but rather `*mut hoedown_document` and `*mut c_void` are the same types. In Rust it's more idiomatic to have those be separate types (to get at least *some* level of type safety). Creating an empty enum (`enum hoedown_document {}`) should serve this purpose